pkg/provisioners/channel_util.go

```diff
@@ -64,29 +65,40 @@ func CreateK8sService(ctx context.Context, client runtimeClient.Client, c *event
 func getVirtualService(ctx context.Context, client runtimeClient.Client, c *eventingv1alpha1.Channel) (*istiov1alpha3.VirtualService, error) {
 	vsk := runtimeClient.ObjectKey{
 		Namespace: c.Namespace,
-		Name:      controller.ChannelVirtualServiceName(c.ObjectMeta.Name),
+		Name:      ChannelVirtualServiceName(c.ObjectMeta.Name),
 	}
 	vs := &istiov1alpha3.VirtualService{}
 	err := client.Get(ctx, vsk, vs)
 	return vs, err
 }
 
-func CreateVirtualService(ctx context.Context, client runtimeClient.Client, c *eventingv1alpha1.Channel) (*istiov1alpha3.VirtualService, error) {
-	virtualService, err := getVirtualService(ctx, client, c)
+func CreateVirtualService(ctx context.Context, client runtimeClient.Client, channel *eventingv1alpha1.Channel) (*istiov1alpha3.VirtualService, error) {
+	virtualService, err := getVirtualService(ctx, client, channel)
 
 	// If the resource doesn't exist, we'll create it
 	if errors.IsNotFound(err) {
-		virtualService = newVirtualService(c)
+		virtualService = newVirtualService(channel)
 		err = client.Create(ctx, virtualService)
+		if err != nil {
+			return nil, err
+		}
+		return virtualService, nil
 	}
-
-	// If an error occurs during Get/Create, we'll requeue the item so we can
-	// attempt processing again later. This could have been caused by a
-	// temporary network failure, or any other transient reason.
 	if err != nil {
 		return nil, err
 	}
 
+	// Update VirtualService if it has changed. This is possible since in version 0.2.0, the destinationHost in
+	// spec.HTTP.Route for the dispatcher was changed from *-clusterbus to *-dispatcher. Even otherwise, this
+	// reconciliation is useful for the future mutations to the object.
+	expected := newVirtualService(channel)
```

Comment:
Let's make the same change to K8s `Service`s as well (comparing and updating if needed).

Reply:
Hmm, I did this but had to tweak a bit since `service.spec.clusterIP` is immutable. Take a look and let me know.